src/app/contexts/UserContext/index.test.tsx

```diff
@@ -61,4 +64,77 @@ describe('UserContext', () => {
       {},
     );
   });
+  describe('ckns_mvt cookie', () => {
+    const cookieSetterSpy = jest.spyOn(Cookie, 'set');
+    const cookieGetterSpy = jest.spyOn(Cookie, 'get');
+    const isOperaProxySpy = jest.spyOn(isOperaProxy, 'default');
+    const onClientSpy = jest.spyOn(onClient, 'default');
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+      Cookie.remove('ckns_mvt');
+    });
+
+    it('should call cookie logic when not opera mini and is on client', () => {
+      onClientSpy.mockImplementationOnce(() => true as unknown as Location);
+      isOperaProxySpy.mockImplementationOnce(() => false);
+
+      render(<DummyComponentWithContext />);
+
+      expect(cookieGetterSpy).toHaveBeenCalled();
+    });
+
+    it('should not call cookie logic when on opera mini and is on client', () => {
+      onClientSpy.mockImplementationOnce(() => true as unknown as Location);
+      isOperaProxySpy.mockImplementationOnce(() => true);
+
+      render(<DummyComponentWithContext />);
+
+      expect(cookieGetterSpy).not.toHaveBeenCalled();
+    });
+
+    it('should not call cookie logic when not on opera mini and is not on client', () => {
+      onClientSpy.mockImplementationOnce(() => false);
+      isOperaProxySpy.mockImplementationOnce(() => false);
+
+      render(<DummyComponentWithContext />);
+
+      expect(cookieGetterSpy).not.toHaveBeenCalled();
+    });
+
+    it('should not set cookie when ckns_mvt cookie already exists', () => {
+      onClientSpy.mockImplementationOnce(() => true as unknown as Location);
+      isOperaProxySpy.mockImplementationOnce(() => false);
+      Cookie.set('ckns_mvt', 'foo');
+      cookieSetterSpy.mockClear();
```

Comment:
I think we should remove this
```suggestion
```

Reply:
I think we need this, as this without it, this expect would fail https://github.com/bbc/simorgh/blob/7cf56575dbf023a85c8fac761882f5a38ddd5250/src/app/contexts/UserContext/index.test.tsx#L114, as we've done this manually for this test case